packages/ui/src/components/Channel/index.tsx

```diff
@@ -279,169 +283,94 @@ export function Channel({
-                                    ? setShowAddGalleryPost(true)
-                                    : setShowBigInput(true)
+                                channelType={channel.type}
+                                channelId={channel.id}
+                                firstUnreadId={
+                                  initialChannelUnread?.countWithoutThreads ??
+                                  0 > 0
+                                    ? initialChannelUnread?.firstUnreadPostId
+                                    : null
+                                }
+                                unreadCount={
+                                  initialChannelUnread?.countWithoutThreads ?? 0
                                 }
-                                icon={
-                                  <Icon
-                                    type="Add"
-                                    size={'$s'}
-                                    marginRight={'$s'}
-                                  />
+                                onPressPost={
+                                  isChatChannel ? undefined : goToPost
                                 }
+                                onPressReplies={goToPost}
+                                onPressImage={goToImageViewer}
+                                onEndReached={onScrollEndReached}
+                                onStartReached={onScrollStartReached}
+                                onPressRetry={onPressRetry}
+                                onPressDelete={onPressDelete}
+                                activeMessage={activeMessage}
+                                setActiveMessage={setActiveMessage}
+                                ref={flatListRef}
                               />
                             )}
                           </View>
                         )}
-                        {!negotiationMatch && isChatChannel && canWrite && (
-                          <NegotionMismatchNotice />
-                        )}
-                        {channel.isDmInvite && (
-                          <DmInviteOptions channel={channel} goBack={goBack} />
-                        )}
-                        {!negotiationMatch && isChatChannel && canWrite && (
-                          <NegotionMismatchNotice />
-                        )}
-                        {channel.type === 'gallery' && canWrite && (
-                          <AddGalleryPost
-                            showAddGalleryPost={showAddGalleryPost}
-                            setShowAddGalleryPost={setShowAddGalleryPost}
-                            setShowGalleryInput={setShowBigInput}
-                            onSetImage={handleGalleryImageSet}
-                          />
-                        )}
-                      </YStack>
-                    </KeyboardAvoidingView>
+                      </AnimatePresence>
+
+                      {canWrite && (
```

Comment:
in case you saw my post in Homestead about similar code and are confused where it went: I had a nice switch statement here that I posted in Homestead in which I moved this logic to a `useMemo` producing an `inputAccessoryViewType` string, and `switch`ed on that value here. I used that code in a previous attempt at unifying safe area insets, but changed direction; and I think this code, although more verbose, is more flexible and direct.